### PR TITLE
Implement Update user API

### PR DESCRIPTION
**Pull request by utuedey**
## Description
Develop API endpoint to update a user

## Acceptance Criteria
- The endpoint should be accessible at `/api/v1/users/:userId`
- The endpoint should accept HTTP `PATCH` requests.
- The endpoint should be secured to ensure that only authenticated users can update user
- The endpoint should check that the user has the appropriate permissions
- The endpont should validate all data passed in the body of the request
- Requests to the endpoint must include a valid authentication token in the Authorization header. `Authorization: Bearer <token>`
- Given a request with a valid user ID, when the user is authenticated and authorized, the system should update the user with a status 
code of 200
- Given a request with an invalid or non-existent user ID, when the user is authenticated and authorised, the system should return a 404 Not Found status.

## Request
- Endpoint
PATCH `/api/v1/users/:userId`

- Request Body
```
{
    "name": "String",
    "phone_number": "String"
}
```
- Parameters:
- `userId`: string

## Expected Outcome
- Successful Response
-  Only Admin can update user details successfully.

```
{
  "status": "success",
  "message": "User Updated Successfully",
  "user": {
    "id": "String",
    "name": "String",
    "phone_number": "String"
  }
}
```
- Error Response:
```
{
  "status":string,
  "message": string,
  "status_code":int
}
```
## Purpose
Enable updating of user information from the user Details Page.

## Requirements
- Develop server-side logic to update a single user
- Ensure compliance with security standards for handling user information.

## Checklist

- [x]  Define API endpoint to update user details.
- [x]  Implement logic to update user information in the database.
- [x]  Handle success and error responses.

## Unittests
- Successful response with valid id
- Invalid userId
- Missing userId
- Invalid request body

### CODE REVIEW

File: pkg/controller/user/user.go [hunk 1]
```diff
@@ -92,3 +92,36 @@ func (base *Controller) LoginUser(c *gin.Context) {
...
+	// Validate the request data
+	if err := base.Validator.Struct(&req); err != nil {
+		rd := utility.BuildErrorResponse(http.StatusUnprocessableEntity, "error", "Validation failed", utility.ValidationResponse(err, base.Validator), nil)
+		c.JSON(http.StatusUnprocessableEntity, rd)
+		return
+	}
+
+	// Extract user ID from the URL parameter
+	userID := c.Param("userId")
+
```
Comment (NwokoyeChigozie) [inline pkg/controller/user/user.go:116]: move this up, should be one of the first things you do in this function

File: services/user/user.go [hunk 2]
```diff
@@ -155,3 +161,45 @@ func LoginUser(req models.LoginRequestModel, db *gorm.DB) (gin.H, int, error) {
...
+
+	// convert userID to UUID
+	id, err := uuid.Parse(userID)
+	if err != nil {
+		return responseData, http.StatusNotFound, fmt.Errorf("invalid user ID format")
+	}
+	// Check if the user exists
+	if err := db.Where("id = ?", id).First(&user).Error; err != nil {
+		return responseData, http.StatusNotFound, fmt.Errorf("user not found")
+	}
```
Comment (NwokoyeChigozie) [inline services/user/user.go:180]: please follow what I told others to do about model to db interaction

File: services/user/user.go [hunk 3]
```diff
@@ -155,3 +161,45 @@ func LoginUser(req models.LoginRequestModel, db *gorm.DB) (gin.H, int, error) {
...
+
+	// Update the user data
+	user.Name = req.Name
+	user.PhoneNumber = req.PhoneNumber
+
+	// Save the updated user data
+	if err := db.Save(&user).Error; err != nil {
+		return responseData, http.StatusInternalServerError, fmt.Errorf("failed to update user: %v", err)
+	}
```
Comment (NwokoyeChigozie) [inline services/user/user.go:194]: follow what already exists

**Review by NwokoyeChigozie — CHANGES_REQUESTED** [review]

File: pkg/controller/user/user.go (hunk 1 above)
Comment (utuedey) [reply pkg/controller/user/user.go:116]: Noted sir, working on it.

**Pull request closed without merging**